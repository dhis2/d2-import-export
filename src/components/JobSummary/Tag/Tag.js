import React from 'react'
import PropTypes from 'prop-types'

import s from './Tag.module.css'

const Tag = ({ text, success, warning, error, info }) => {
    const classNames = [
        s.container,
        warning && s.warning,
        error && s.error,
        success && s.success,
        info && s.info,
    ]
        .filter(a => !!a)
        .join(' ')

    return (
        <div className={classNames}>
            <span className={s.text}>{text}</span>
        </div>
    )
}

Tag.propTypes = {
    text: PropTypes.string.isRequired,
    error: PropTypes.bool,
    info: PropTypes.bool,
    success: PropTypes.bool,
    warning: PropTypes.bool,
}

export { Tag }
