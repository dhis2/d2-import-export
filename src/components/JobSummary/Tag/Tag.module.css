.container {
    display: inline-flex;
    -moz-box-align: center;
    align-items: center;
    height: 32px;
    margin: 4px;
    /* border-radius: 16px; */
    background-color: rgb(243, 245, 247);
    font-size: 14px;
    line-height: 16px;
    color: rgb(33, 41, 52);
}

.text {
    margin: 0px 12px;
}

.success {
    background-color: rgb(0, 121, 107);
    color: rgb(255, 255, 255);
    font-weight: 500;
}

.error {
    background-color: rgb(185, 36, 43);
    color: rgb(255, 255, 255);
    font-weight: 500;
}

.warning {
    background-color: rgb(255, 147, 2);
    color: rgb(255, 255, 255);
    font-weight: 500;
}

.info {
    background-color: rgb(33, 41, 52);
    color: rgb(255, 255, 255);
    font-weight: 500;
}
