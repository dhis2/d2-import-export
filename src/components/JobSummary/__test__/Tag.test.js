import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect'

import { Tag } from '../Tag'

const propMake = type => ({
    [type]: true,
    text: `${type} message`,
})

describe('different tag types', () => {
    const allTypes = ['success', 'warning', 'error', 'info']

    allTypes.forEach(t =>
        it(`matches snapshot - ${t}`, () => {
            const { asFragment } = render(<Tag {...propMake(t)} />)
            expect(asFragment()).toMatchSnapshot()
        })
    )
})
