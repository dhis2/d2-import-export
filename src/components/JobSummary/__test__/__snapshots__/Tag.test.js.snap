// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`different tag types matches snapshot - error 1`] = `
<DocumentFragment>
  <div
    class="container error"
  >
    <span
      class="text"
    >
      error message
    </span>
  </div>
</DocumentFragment>
`;

exports[`different tag types matches snapshot - info 1`] = `
<DocumentFragment>
  <div
    class="container info"
  >
    <span
      class="text"
    >
      info message
    </span>
  </div>
</DocumentFragment>
`;

exports[`different tag types matches snapshot - success 1`] = `
<DocumentFragment>
  <div
    class="container success"
  >
    <span
      class="text"
    >
      success message
    </span>
  </div>
</DocumentFragment>
`;

exports[`different tag types matches snapshot - warning 1`] = `
<DocumentFragment>
  <div
    class="container warning"
  >
    <span
      class="text"
    >
      warning message
    </span>
  </div>
</DocumentFragment>
`;